LanguageFeatures/Extension-types/static_analysis_extension_types_A08_t06.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We say that a type T is incompatible with await if at least one
+/// of the following criteria holds:
+/// - T is an extension type that does not implement Future.
+/// - T is S?, and S is incompatible with await.
+/// - T is X & B, and either:
+/// - B is incompatible with await, or
+/// - B does not derive a future type, and X is incompatible with await.
+/// - T is a type variable with bound S, and S is incompatible with await.
+/// Consider an expression of the form await e. A compile-time error occurs if
+/// the static type of e is incompatible with await.
+///
+/// @description Checks that it is a compile-time error if `await e` occurs, and
+/// the static type of `e` is `X & B`, and `B` is incompatible with await.
```

Comment:
We don't have any type variables here as far as I can see, so we could never have a type of the form `X & B`. We'd need something like `void foo<X extends ET>(X x)`, and then promote, and then `await x;`.